client/ayon_maya/plugins/publish/extract_layout.py

```diff
@@ -176,6 +180,32 @@ def process(self, instance):
         self.log.debug("Extracted instance '%s' to: %s",
                        instance.name, json_representation)
 
+    def get_containers_sub_assembly(self, containers, container_root):
+        """Allow to collect the asset containers through sub-assembly workflow if
+        there is a layout container
+
+        Args:
+            containers (list): Ayon asset containers
+            container_root (str): the root of the asset container
+
+        Returns:
+            dict: container dict with the container and its related asset name
+        """
+        all_containers_set = {}
+        for existing_container in containers:
+            if cmds.getAttr(f"{existing_container}.loader").lower() == "layoutloader":
+                for container in cmds.sets(existing_container, query=True):
```

Comment:
Technically not everything in the first container, may itself be a container (or even an object set for that matter).
With a loaded layout product 'most likely it is' but artists really break everything. So I'd at least filter to the data that we'd expect before doing anything with it.

Also, this initial set query - could also be replaced by `get_container_members`
```suggestion
                for container in get_container_members(existing_container):
                    # todo: confirm the object is actually a container?
```